### PR TITLE
[CIS-503] Move `name` and `imageURL` to models

### PR DESCRIPTION
**Pull request by b-onc**
Resulting work of 2 days of effort

### CODE REVIEW

**Review by evsaev — COMMENTED** [review]
Does adding `name` and `imageURL` to user/channel allow us to remove all references to `NameAndImageExtraData`?

File: Sources_v3/StreamChat/Models/Member.swift [hunk 1]
```diff
@@ -12,7 +12,7 @@ import Foundation
 ///
 /// Learn more about using custom extra data in our [cheat sheet](https://github.com/GetStream/stream-chat-swift/wiki/StreamChat-SDK-Cheat-Sheet#working-with-extra-data).
 ///
-public typealias ChatChannelMember = _ChatChannelMember<NameAndImageExtraData>
+public typealias ChatChannelMember = _ChatChannelMember<DefaultExtraData.User>
```
Comment (VojtaStavik) [inline Sources_v3/StreamChat/Models/Member.swift:15]: This was actually wrong all the time, right? :)

**Review by VojtaStavik — APPROVED** [review]
Great 🎉 Let's see what breaks :)

File: Sources_v3/StreamChat/Models/Member.swift (hunk 1 above)
Comment (b-onc) [reply Sources_v3/StreamChat/Models/Member.swift:15]: yep, in many places we've used `NameAndImageExtraData` where we should've used `DefaultExtraData.User`, I've refactored them all (but only for Channel and User, obv)

**Comment by b-onc** [comment]
@evsaev yes!

**Comment by VojtaStavik** [comment]
@b-onc This means we can get rid of `UIExtraData` too, right? 🎉

**Comment by evsaev** [comment]
> @evsaev yes!

@b-onc hah, I somehow oversaw the commit with `Remove NameAndImageExtraData` message 🤦 Nice job!

**Comment by b-onc** [comment]
@VojtaStavik yep, it'll come in a separate PR though

**Pull request merged**